pilot/pkg/model/validation.go

```diff
@@ -1529,6 +1557,60 @@ func ValidateAuthorizationPolicy(name, namespace string, msg proto.Message) erro
 	return nil
 }
 
+// isFirstClassFieldEmpty return false if there is at least one first class field (e.g. properties)
+func isFirstClassFieldEmpty(subject *rbac.Subject) bool {
+	return len(subject.User) == 0 && len(subject.Group) == 0 && len(subject.Properties) == 0 &&
+		len(subject.Namespaces) == 0 && len(subject.NotNamespaces) == 0 && len(subject.Groups) == 0 &&
+		len(subject.NotGroups) == 0 && len(subject.Ips) == 0 && len(subject.NotIps) == 0 &&
+		len(subject.Names) == 0 && len(subject.NotNames) == 0
+}
+
+// Returns true if the user defines a property that already exists in the first-class fields, false
+// if none has overlapped.
+// In the future when we introduce expression conditions in properties field, we might need to revisit
+// this function.
+func hasExistingFirstClassFieldInBinding(propertiesKey string, subject *rbac.Subject) bool {
+	// Cannot use attributes from authz package since there is a import cycle. However, these constants can be
+	// defined at another place and both authz and model package can access them.
+	const (
+		// Same as attributes from authz package.
+		attrSrcPrincipal       = "source.principal"
+		attrRequestClaimGroups = "request.auth.claims[groups]"
+		attrSrcNamespace       = "source.namespace"
+		attrSrcIP              = "source.ip"
+	)
+	switch propertiesKey {
+	case attrSrcPrincipal:
+		if len(subject.Names) > 0 {
```

Comment:
Why not just `return len(subject.Names) > 0`

Reply:
Good catch! Thanks!